docs/styleguide/PyGuide.md

```diff
@@ -0,0 +1,821 @@
+## 2 Python Language Rules 
+
+<a id="s2.1-pylint"></a>
+<a id="21-pylint"></a>
+<a id="pylint"></a>
+### 2.1 PyLint 
+
+Run [pylint](https://github.com/PyCQA/pylint) over your code using this [pylintrc](../../.pylintrc).
+
+- Every warning reported by [pylint](https://github.com/PyCQA/pylint) must be resolved in one of the following way:
+  - *Preferred solution*: Change the code to fix the warning.
+  - *Exception*: Suppress the warning if they are inappropriate so that other issues are not hidden. 
+    To suppress warnings you can set a line-level comment
+    ```python
+    dict = 'something awful'  # Bad Idea... pylint: disable=redefined-builtin
+    ```
+    or update [pylintrc](../../.pylintrc) if applicable for the whole project. If the reason for the suppression 
+    is not clear from the symbolic name, add an explanation.
+      
+
+<a id="s2.2-imports"></a>
+<a id="22-imports"></a>
+<a id="imports"></a>
+### 2.2 Imports 
+
+- Use absolute imports, as they are usually more readable and tend to be better behaved
+  ```python
+  # Correct:
+  import mypkg.sibling
+  from mypkg import sibling
+  from mypkg.sibling import example
+  ```
+  ```python
+  # Wrong:
+  from . import sibling
+  from .sibling import example
+  ```
+- Imports should usually be on separate lines:
+  ```python
+  # Correct:
+  from nncf.api.compression import CompressionLoss
+  from nncf.api.compression import CompressionScheduler
+  from nncf.api.compression import CompressionAlgorithmBuilder
+  from nncf.api.compression import CompressionAlgorithmController
+  ```
+  ```python
+  # Wrong:
+  from nncf.api.compression import CompressionLoss, CompressionScheduler, CompressionAlgorithmBuilder, \
+      CompressionAlgorithmController
+  ```
+- Imports are always put at the top of the file, just after any module comments and docstrings, and before module 
+  globals and constants. Imports should be grouped in the following order:
+  - Standard library imports.
+  - Related third party imports. 
+  - Local application/library specific imports.
+  
+  You should put a blank line between each group of imports.
+- When importing a class from a class-containing module, the preferred behavior is as follows:
+  ```python
+  from myclass import MyClass
+  from foo.bar.yourclass import YourClass
+  ```
+  If this spelling causes local name clashes, then spell them explicitly:
+  ```python
+  import myclass
+  import foo.bar.yourclass
+  ```
+  and use `myclass.MyClass` and `foo.bar.yourclass.YourClass`.
+
+- Wildcard imports (`from module import *`) should be avoided, as they make it unclear which names are present 
+  in the namespace, confusing both readers and many automated tools.
+
+<a id="s2.3-3rd-party-packages"></a>
+<a id="23-3rd-party-packages"></a>
+<a id="3rd-party-packages"></a>
+### 2.3 3rd party packages 
+
+Do not add new third-party dependencies unless absolutely necessary. All things being equal, give preference to built-in packages.  
```

Comment:
I guess we should oblige people to provide a full list of code borrwings/new dependencies in the PR description when they introduce new PRs.

Reply:
What do you think If I add this to the contribution guide?